misk-hibernate/src/test/kotlin/misk/hibernate/SchemaMigratorTest.kt

```diff
@@ -110,31 +122,52 @@ internal class SchemaMigratorTest {
     assertThat(tableExists("table_2")).isFalse()
     assertThat(tableExists("table_3")).isFalse()
     assertThat(tableExists("table_4")).isFalse()
+    assertThat(tableExists("library_table")).isFalse()
+    assertThat(tableExists("library_table_2")).isFalse()
 
     // When we apply migrations, the table is present and contains the applied migrations.
-    schemaMigrator.applyAll("SchemaMigratorTest", setOf())
-    assertThat(schemaMigrator.appliedMigrations()).containsExactly(1001, 1002)
+    schemaMigrator.applyAll("SchemaMigratorTest", sortedSetOf())
+    assertThat(schemaMigrator.appliedMigrations()).containsExactly(
+        NamedspacedMigration(1001),
+        NamedspacedMigration(1002),
+        NamedspacedMigration("name/space/", 1001))
     assertThat(tableExists("schema_version")).isTrue()
     assertThat(tableExists("table_1")).isTrue()
     assertThat(tableExists("table_2")).isTrue()
+    assertThat(tableExists("library_table")).isTrue()
     assertThat(tableExists("table_3")).isFalse()
     assertThat(tableExists("table_4")).isFalse()
+    assertThat(tableExists("library_table_2")).isFalse()
     schemaMigrator.requireAll()
 
     // When new migrations are added they can be applied.
-    resourceLoader.put("${config.data_source.migrations_resource}/v1003__movies.sql", """
+    resourceLoader.put("${mainSource}/v1003__movies.sql", """
         |CREATE TABLE table_3 (name varchar(255))
         |""".trimMargin())
-    resourceLoader.put("${config.data_source.migrations_resource}/v1004__movies.sql", """
+    resourceLoader.put("${mainSource}/v1004__movies.sql", """
         |CREATE TABLE table_4 (name varchar(255))
         |""".trimMargin())
-    schemaMigrator.applyAll("SchemaMigratorTest", setOf(1001, 1002))
-    assertThat(schemaMigrator.appliedMigrations()).containsExactly(1001, 1002, 1003, 1004)
+    resourceLoader.put("${mainSource}/namespace/v1001__props.sql", """
+        |CREATE TABLE library_table_2 (name varchar(255))
+        |""".trimMargin())
+    schemaMigrator.applyAll("SchemaMigratorTest", sortedSetOf(
+        NamedspacedMigration(1001),
+        NamedspacedMigration(1002),
+        NamedspacedMigration("name/space/", 1001)))
+    assertThat(schemaMigrator.appliedMigrations()).containsExactly(
+        NamedspacedMigration(1001),
+        NamedspacedMigration(1002),
+        NamedspacedMigration(1003),
+        NamedspacedMigration(1004),
+        NamedspacedMigration("name/space/", 1001),
```

Comment:
does the namespace provide any value other than allowing migrations with the same name to avoid conflicts?

Reply:
No, it's just libraries and projects can keep their own independent version 1..n without copying or conflicts